mitodl/mitxonline
# Adds explicit binding of flexible price request forms to courseware objects

#### Pre-Flight checklist


#### What are the relevant tickets?

#724 

#### What's this PR do?

Adds a couple of fields and relations to the FlexiblePriceForm to link them directly to a Course or Program object. Also updates the submit logic to retrieve the correct courseware object based on some rules. 

The courseware object can be retrieved through the `get_parent_courseware` function (was `get_parent_product`), which returns one of (in order):
- The selected program, if there is one
- The selected course, if there is one
- The parent page's course, if there is one

Prior to this, the model would only look at the form's place in the page hierarchy to determine the courseware object. This allows us to set up a Flexible Pricing request form for a program without having to have program pages in the system. When the system games program pages, the logic here should be updated to pull the parent page's program as well. 

#### How should this be manually tested?

All automated tests should pass.

1. Create or edit a flexible pricing request form.
2. You should see dropdowns for Course and Program. Set one or the other. (If you set both, the Program takes precedence.) 
3. Save the form, and submit a request through the form.
4. The resulting submission should be linked with the appropriate courseware object.
